### PR TITLE
To add RMSProp algorithm documentation

### PR DESCRIPTION
**Pull request by iramazanli**
It has been discussed before that adding description of Optimization algorithms to PyTorch Core documentation may result in a nice Optimization research tutorial. In the following tracking issue we mentioned about all the necessary algorithms and links to the originally published paper  https://github.com/pytorch/pytorch/issues/63236.

In this PR we are adding description of RMSProp to the documentation.  For more details, we refer to the paper   https://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf

RMSProp

### CODE REVIEW

File: torch/optim/rmsprop.py [hunk 1]
```diff
@@ -6,15 +6,34 @@
...
+
+    For further details regarding the algorithm we refer to
+    `lecture notes <https://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf>`_ by G. Hinton.
+    and centered version `Generating Sequences
     With Recurrent Neural Networks <https://arxiv.org/pdf/1308.0850v5.pdf>`_.
-
     The implementation here takes the square root of the gradient average before
     adding epsilon (note that TensorFlow interchanges these two operations). The effective
-    learning rate is thus :math:`\alpha/(\sqrt{v} + \epsilon)` where :math:`\alpha`
+    learning rate is thus :math:`\gamma/(\sqrt{v} + \epsilon)` where :math:`\gamma`
```
Comment (albanD) [inline torch/optim/rmsprop.py:36]: This was a typo in the previous doc right?

File: torch/optim/rmsprop.py [hunk 2]
```diff
@@ -6,15 +6,34 @@
...
     r"""Implements RMSprop algorithm.
 
-    Proposed by G. Hinton in his
-    `course <https://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf>`_.
-
-    The centered version first appears in `Generating Sequences
+    .. math::
+       \begin{aligned}
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\textbf{input}      : \alpha \text{ (alpha)}, \gamma \text{ (lr)},
```
Comment (albanD) [inline torch/optim/rmsprop.py:12]: What about the momentum, weight decay and centered arguments?

File: torch/optim/rmsprop.py [hunk 3]
```diff
@@ -6,15 +6,34 @@
...
+                \theta_0 \text{ (params)}, f(\theta) \text{ (objective)}                         \\
+            &\textbf{initialize} : v_0 \leftarrow 0 \text{ (square average)},
+                t \leftarrow 0                                                            \\[-1.ex]
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\bf{while} \text{  } \theta_t \text{ not converged } \mathbf{do}                    \\
+            &\hspace{5mm}t             \leftarrow   t+1                                          \\
+            &\hspace{5mm}g_t           \leftarrow   \nabla_{\theta} f_t (\theta_{t-1})
```
Comment (albanD) [inline torch/optim/rmsprop.py:19]: This should be `f_{t-1}` no?

**Review by albanD — COMMENTED** [review]
Thanks for the rendered version, that helps a lot!

**Review by albanD — COMMENTED** [review]
The overall structure looks good.

**Review by albanD — APPROVED** [review]
LGTM
